### DISCUSSION TITLE
GpuBufferDescriptor

### DISCUSSION THREAD

**Discussion by mzkn0** [wasm-bindgen/wasm-bindgen#3815] [Q&A] [answered]
Why is the type of the `size` argument of the GpuBufferDescriptor constructor `f64` instead of `u64`?

**Comment by daxpedda** [comment, marked as answer]
I'm not entirely sure of what the original plan was, but generally speaking this is correct.
JS represents all `Number`s as `f64`, even if the IDL specifies it as `unsigned long long` it takes a `Number` in JS and not a `BigInt` (the only true integer type I'm aware of in JS).

I'm not entirely sure if this is a limitation by the spec (because [the spec uses the word "Integer"](https://webidl.spec.whatwg.org/#idl-unsigned-long-long)), or if this is a limitation by JS being used to implement the spec in browsers.

In any case, as long as browsers take a `Number` for `unsigned long long` there isn't much we can do here.

**Reply by mzkn0** [reply]
Thank you for your response. I understand.
